src/main/kotlin/no/fdk/fdk_concept_harvester/tbx/TbxUtils.kt

```diff
@@ -0,0 +1,305 @@
+package no.fdk.fdk_concept_harvester.tbx
+
+import no.fdk.fdk_concept_harvester.Application
+import no.fdk.fdk_concept_harvester.adapter.OrganizationsAdapter
+import org.apache.jena.ext.xerces.util.URI
+import org.apache.jena.rdf.model.Model
+import org.apache.jena.rdf.model.ModelFactory
+import org.apache.jena.rdf.model.RDFNode
+import org.apache.jena.rdf.model.Resource
+import org.apache.jena.vocabulary.*
+import org.slf4j.LoggerFactory
+import org.w3c.dom.Document
+import org.w3c.dom.Element
+import org.w3c.dom.Node
+import org.w3c.dom.NodeList
+import java.io.ByteArrayInputStream
+import javax.xml.parsers.DocumentBuilderFactory
+import javax.xml.xpath.XPath
+import javax.xml.xpath.XPathConstants
+import javax.xml.xpath.XPathFactory
+
+private val logger = LoggerFactory.getLogger(Application::class.java)
+
+private val SKOSNO_NS = "https://data.norge.no/vocabulary/skosno#"
+private val SKOSNO_DEFINISJON = "Definisjon"
+private val SKOSNO_DEFINISJON_PROPERTY = "definisjon"
+
+fun parseTBXResponse(tbxContent: String, rdfSource: String?, orgAdapter: OrganizationsAdapter): Model? {
+    val model = ModelFactory.createDefaultModel()
+        .setPrefixes()
+
+    try {
+        // Parse TBX
+        val doc = DocumentBuilderFactory
+            .newInstance()
+            .newDocumentBuilder()
+            .parse(ByteArrayInputStream(tbxContent.toByteArray()))
+
+        val xpath = XPathFactory.newInstance().newXPath()
+
+        // reading tbxHeader
+        val collectionResource = model.createResource(xpath.collectionUri(doc))
+        val publisherResource = xpath.publisherUri(doc, orgAdapter)?.let { model.createResource(it) }
+
+        model.add(model.createStatement(collectionResource, RDF.type, SKOS.Collection))
+        model.add(
+            model.createStatement(collectionResource, RDFS.label,
+            model.createLiteral(xpath.collectionName(doc), "en")))
+        model.addPublisher(collectionResource, publisherResource)
+
+        // reading conceptEntry
+        xpath.conceptEntries(doc)
+            .forEach { addConceptEntryNodeToModel(it as Element, doc, collectionResource, publisherResource, model) }
+
+    } catch (ex: Exception) {
+        logger.error("Parse from $rdfSource has failed", ex)
+        return null
+    }
+
+    return model
+}
+
+private fun addConceptEntryNodeToModel(conceptEntry: Element, doc: Document, collectionResource: Resource,
+                                       publisher: Resource?, model: Model) {
+    val xpath = XPathFactory.newInstance().newXPath()
+    val conceptUri = xpath.conceptUri(conceptEntry, doc)
+    val langSections = xpath.langSections(conceptEntry, doc)
+
+    val conceptResource = model.createResource(URI(conceptUri).toString())
```

Comment:
Har du ikke allerede verifisert den som uri?
```suggestion
    val conceptResource = model.createResource(conceptUri)
```

Reply:
Jeg gjør dette for å være sikker dette er faktisk en URI

Reply:
Ja, men du har vel allerede gjort det. Produserer conceptUri vha `xpath.conceptUri`, på linje 167 så kjøres jo den uri-sjekken